plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/rules/transport/TransportIndexCorrelationRuleAction.java

```diff
@@ -0,0 +1,234 @@
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.plugin.correlation.rules.action.IndexCorrelationRuleAction;
+import org.opensearch.plugin.correlation.rules.action.IndexCorrelationRuleRequest;
+import org.opensearch.plugin.correlation.rules.action.IndexCorrelationRuleResponse;
+import org.opensearch.plugin.correlation.rules.model.CorrelationRule;
+import org.opensearch.plugin.correlation.utils.CorrelationRuleIndices;
+import org.opensearch.plugin.correlation.utils.IndexUtils;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * Transport Action for indexing correlation rules.
+ *
+ * @opensearch.internal
+ */
+public class TransportIndexCorrelationRuleAction extends HandledTransportAction<IndexCorrelationRuleRequest, IndexCorrelationRuleResponse> {
+
+    private static final Logger log = LogManager.getLogger(TransportIndexCorrelationRuleAction.class);
+
+    private final Client client;
+
+    private final CorrelationRuleIndices correlationRuleIndices;
+
+    private final ClusterService clusterService;
+
+    /**
+     * Parameterized ctor for Transport Action
+     * @param transportService TransportService
+     * @param client OS client
+     * @param actionFilters ActionFilters
+     * @param clusterService ClusterService
+     * @param correlationRuleIndices CorrelationRuleIndices which manages lifecycle of correlation rule index
+     */
+    @Inject
+    public TransportIndexCorrelationRuleAction(
+        TransportService transportService,
+        Client client,
+        ActionFilters actionFilters,
+        ClusterService clusterService,
+        CorrelationRuleIndices correlationRuleIndices
+    ) {
+        super(IndexCorrelationRuleAction.NAME, transportService, actionFilters, IndexCorrelationRuleRequest::new);
+        this.client = client;
+        this.clusterService = clusterService;
+        this.correlationRuleIndices = correlationRuleIndices;
+    }
+
+    @Override
+    protected void doExecute(Task task, IndexCorrelationRuleRequest request, ActionListener<IndexCorrelationRuleResponse> listener) {
+        AsyncIndexCorrelationRuleAction asyncAction = new AsyncIndexCorrelationRuleAction(request, listener);
+        asyncAction.start();
+    }
+
+    class AsyncIndexCorrelationRuleAction {
```

Comment:
nit: make private if we're not using it outside the package?

Reply:
made this `private`.